docs/content/sips/008-using-oci-registries.md

```diff
@@ -0,0 +1,364 @@
+          "digest": "sha256:a4699e4f9ef3f4922f38f0d017aa26438908f38caf020a739e0ee27fe796eb02",
+          "path": "my-file.json"
+        }
+      ]
+    }
+  ]
+}
+```
+
+This is all the information required for Spin to be able to push, pull, then run
+an application from an OCI registry.
+
+### `spin oci push`
+
+`spin oci push` is intended to give users the ability to distribute their
+application using widely available container registry services, giving them
+as much flexibility as possible in order to integrate `spin oci push` and `spin up`
+into their *existing* workflows. To this end, it is intended to be as unopinionated
+as possible when it comes to versioning and tag mutability.
+
+As a result, `spin oci push` should allow the option to accept a user-defined
+name and tag for the artifact pushed to the registry:
+
+```bash
+$ spin oci push --file <path to spin.toml> myregistry.com/myusername/myapp:v1
+# OR
+$ spin oci push --file <path to spin.toml> myregistry.com/myusername/myapp:latest
+# OR
+$ spin oci push --file <path to spin.toml> myregistry.com/myusername/myapp:v0.1.0+r2d2
+```
+
+However, it should also preserve deriving the reference and tag from the `spin.toml`
+application name and version. For example, for the following `spin.toml`:
+
+```toml
+name = "myregistry.com/myusername/myapp"
+version = "1.2.3"
+```
+
+Running `spin oci push` should result in the application being pushed without
+having to specify the same information on the command line again:
+
+```bash
+$ spin oci push --file <path to spin.toml>
+... Pushed the application to myregistry.com/myusername/myapp:v1.2.3
+# OR
+$ spin oci push --file <path to spin.toml> --buildinfo
+... Pushed the application to myregistry.com/myusername/myapp:v1.2.3+r2d2
+```
+
+Note: deriving the reference and tag from `spin.toml` when no explicit value is passed
+on the command line means the application name must contain the fully qualified
+reference, and the tag can only be a semantic version.
+
+### Impact to `spin deploy`
+
+While `spin oci push` should offer the most flexibility when pushing and tagging
+an application, `spin deploy` is part of an opinionated workflow which could
+continue to enforce its tag mutability and versioning strategy.
+
+In line with the gradual approach to this change, `spin deploy` will also change
+to publish the application to OCI registries, and by the time this change
+will be default, Fermyon Platform and Fermyon Cloud will support this change.
+
+The goal is to have at least one minor release of Spin with `spin oci` before
+updating the *default* `spin deploy` behavior to expect an OCI registry.
+Once Fermyon Platform or Fermyon Cloud support accepting OCI references,
+the intention is to implement `spin deploy --oci` functionality before
+updating the default behavior.
+
+### Implementation status
+
+The `spin oci push`, `spin oci pull`, and `spin oci run` commands are currently
+implemented in this prototype. The implementation uses the
```

Comment:
Ideally link the PR that contains the prototype since it's no longer part of this one.

Reply:
Good catch, updated.